### DISCUSSION TITLE
Synced Blocks

### DISCUSSION THREAD

**Discussion by erichosick** [mdx-js/mdx#2325] [Ideas] [answered]
Is there any existing work or thoughts around supporting synced blocks (an example of this is [Notion Synced Blocks](https://www.notion.so/help/synced-blocks)) in MDX?

**Comment by ChristianMurphy** [comment]
Welcome @erichosick! 👋 

> Is there any existing work or thoughts around supporting synced blocks (an example of this is [Notion Synced Blocks](https://www.notion.so/help/synced-blocks)) in MDX?

No, and I'm not sure how it would fit.
MDX is a language, not a content management system (CMS), nor a collection of React components.
Synced blocks are a React component, not a syntax/language feature, and synced blocks primarily are CMS backed/focused.

You could create a synced block React component, connect that with your CMS of choice, and pass that into MDX today.

**Reply by ChristianMurphy** [reply, marked as answer]
Independently of the concept of synced, MDX also supports `import`ing an MDX file into another.
So it is also possible to create shared snippets that way.

**Reply by erichosick** [reply]
Thank you very much for your thoughts and ideas. Embedding MDX in MDX feels like the best approach to this.
